### ISSUE TITLE
Installation configuration stack

### ISSUE THREAD

**Issue by nscpro** [LibreTime/libretime#1077]
Next error what i see chrome inspector. And Rabbit Setup loading and loading

:8888/setup/setup-functions.php?obj=RabbitMQSetup:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)

I have installed PHP version 7.3.20 and using Ubuntu 18.04.4 LTS

Some details in installation:

Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.
Package raven/raven is abandoned, you should avoid using it. Use sentry/sentry instead.

-----------------------------------------------------
              * Configuring PHP in Apache *          
-----------------------------------------------------
Module rewrite already enabled
Considering dependency mpm_prefork for php7.2:
Considering conflict mpm_event for mpm_prefork:
Considering conflict mpm_worker for mpm_prefork:
Module mpm_prefork already enabled
Considering conflict php5 for php7.2:
Module php7.2 already enabled

**Comment by paddatrapper** [comment]
> `:8888/setup/setup-functions.php?obj=RabbitMQSetup:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)`

What is the content of the RabbitMQ log?

> Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.
> Package raven/raven is abandoned, you should avoid using it. Use sentry/sentry instead.

Known issue - #777

**Comment by nscpro** [comment]
> > `:8888/setup/setup-functions.php?obj=RabbitMQSetup:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)`
> 
> What is the content of `/var/log/airtime/zend.log`?
> 
> > Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.
> > Package raven/raven is abandoned, you should avoid using it. Use sentry/sentry instead.
> 
> Known issue - #777

cat: /var/log/airtime/zend.log: No such file or directory

**Comment by paddatrapper** [comment]
Do you have any extensions installed in your browser that could be blocking Javascript?

**Comment by nscpro** [comment]
No. You can try. If you can install that's good, but i'm not. http://saareobu.eu:8888

**Comment by paddatrapper** [comment]
Can you describe your setup? Are you using a reverse proxy, do you have anything else installed on the box, etc?

**Comment by nscpro** [comment]
i installed libretime uses git clone and command ./install -fiap. Yes i have install DirectAdmin on my VPS.

**Comment by paddatrapper** [comment]
> http://saareobu.eu:8888

I cannot access this

Did the install successfully get to the end (I.e. did you see a prompt to access the site from a browser)? How did you do the port change from 80 to 8888?

**Comment by nscpro** [comment]
i change 80 port to 8888 /etc/apache2/sites-available/airtime.conf file and ports.conf file

**Comment by paddatrapper** [comment]
What are the contents of `/etc/airtime.conf`?

**Comment by nscpro** [comment]
root@mail:/etc/apache2# cat /etc/apache2/sites-available/airtime.conf


<VirtualHost *:8888>
   ServerAdmin foo@bar.org
   DocumentRoot /usr/share/airtime/php/airtime_mvc/public/
   php_admin_value upload_tmp_dir /tmp
   php_value post_max_size 500M
   php_value upload_max_filesize 500M
   php_value request_order "GPC"
   php_value session.gc_probability 0
   php_value session.auto_start 0

   AddOutputFilterByType DEFLATE text/plain
   AddOutputFilterByType DEFLATE text/html
   AddOutputFilterByType DEFLATE text/xml
   AddOutputFilterByType DEFLATE text/css
   AddOutputFilterByType DEFLATE application/xml
   AddOutputFilterByType DEFLATE application/xhtml+xml
   AddOutputFilterByType DEFLATE application/rss+xml
   AddOutputFilterByType DEFLATE application/javascript
   AddOutputFilterByType DEFLATE application/x-javascript
   AddOutputFilterByType DEFLATE application/json


  <Directory /usr/share/airtime/php/airtime_mvc/public/>
      RewriteEngine On
      RewriteCond %{REQUEST_FILENAME} -s [OR]
      RewriteCond %{REQUEST_FILENAME} -l [OR]
      RewriteCond %{REQUEST_FILENAME} -d
      RewriteRule ^.*$ - [NC,L]
      RewriteRule ^.*$ index.php [NC,L]

      DirectoryIndex index.php
      AllowOverride all
      Order allow,deny
      Allow from all

      Require all granted
  </Directory>
</VirtualHost>
root@mail:/etc/apache2#

**Comment by paddatrapper** [comment]
`/etc/airtime.conf`, not the Apache config. Could be in `/etc/airtime/`, I can't remember off-hand

**Comment by nscpro** [comment]
if i look /etc/airtime folder, i see only "airtime.conf.temp" file not airtime.conf file

**Comment by paddatrapper** [comment]
What is the content of that?

**Comment by nscpro** [comment]
root@mail:/etc/airtime# cat airtime.conf.temp


[general]
api_key =
web_server_user = www-data
base_url = localhost
base_port = 80
base_dir = /
cache_ahead_hours = 1
airtime_dir =
station_id =
auth = local


[database]
host = localhost
dbname = airtime
dbuser = airtime
dbpass = airtime


[rabbitmq]
host = 127.0.0.1
port = 5672
user = airtime
password = airtime
vhost = /airtime


[current_backend]
storage_backend=file


[monit]
user =
password =



[pypo]
api_client = 'airtime'
cache_dir = '/var/tmp/airtime/pypo/cache/'
file_dir = '/var/tmp/airtime/pypo/files/' 
tmp_dir = '/var/tmp/airtime/pypo/tmp/'
cache_base_dir = '/var/tmp/airtime/pypo'
log_base_dir = '/var/log/airtime'
pypo_log_dir = '/var/log/airtime/pypo'
liquidsoap_log_dir = '/var/log/airtime/pypo-liquidsoap'
ls_host = '127.0.0.1'
ls_port = '1234'
poll_interval = 3600
push_interval = 1
cue_style = 'pre'
record_bitrate = 256
record_samplerate = 44100
record_channels = 2
record_sample_size = 16
record_file_type = 'ogg'
base_recorded_files = '/var/tmp/airtime/show-recorder/'


[soundcloud]
connection_retries = 3
time_between_retries = 60
soundcloud_client_id = 0
soundcloud_client_secret = 0
soundcloud_redirect_uri = http://libretime.example.org/soundcloud_callback.php
[facebook]
facebook_app_id = 0
facebook_app_url = http://example.org
facebook_app_api_key = 0

[ldap]
hostname = ldap.example.org
binddn = 'uid=libretime,cn=sysaccounts,cn=etc,dc=int,dc=example,dc=org'
password = hackme
account_domain = INT.EXAMPLE.ORG
basedn = 'cn=users,cn=accounts,dc=int,dc=example,dc=org'
filter_field = uid
groupmap_guest = 'cn=guest,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_host = 'cn=host,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_program_manager = 'cn=program_manager,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_admin = 'cn=admins,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_superadmin = 'cn=superadmin,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
root@mail:/etc/airtime#

**Comment by paddatrapper** [comment]
I think you are running into issues with the default port changing. Try run through the setup completely using port 80 and then update the apache config and airtime.conf (which the installer will create)

**Comment by nscpro** [comment]
yeah, but how can i create that, if directadmin uses port 80

**Comment by paddatrapper** [comment]
You can write the file directly - rename the .temp file and change the values. Once that is done, create the media directory with the correct permissions (user and group `libretime`). That way you bypass the installer

**Comment by nscpro** [comment]
i must be change airtime.conf.temp to airtime.conf and change port 8888?

**Comment by nscpro** [comment]
okay i change temp to conf and add port 888 and login admin/admin. Something is wrong. I see broken page..some function or something not work correctly 
![Screen Shot 2020-08-03 at 17 23 46](https://user-images.githubusercontent.com/16063685/89193213-55e02480-d5ae-11ea-81a8-8c1533f5abe7.png)

**Comment by paddatrapper** [comment]
Restart all the services:

```
sudo systemctl restart libretime-*
```
And then see if they run correctly (`systemctl status libretime-{playout,celery,liquidsoap,analyzer}`)

**Comment by nscpro** [comment]
Maybe you find here something, why this not working
![Screen Shot 2020-08-03 at 18 09 03](https://user-images.githubusercontent.com/16063685/89197559-9478dd80-d5b4-11ea-9bed-4d1a86a15eae.png)

**Comment by nscpro** [comment]
![Screen Shot 2020-08-03 at 18 32 38](https://user-images.githubusercontent.com/16063685/89199803-c8a1cd80-d5b7-11ea-8bbd-3f994cd1c5ec.png)

**Comment by paddatrapper** [comment]
The biggest issue I see from the `systemctl` command is that liquidsoap isn't able to correctly communicate with the server. Have you set the API key in the config file? Also what do you see in the Stream Settings page on the site?

**Comment by nscpro** [comment]
No, i don't have API key added a config file. All pages are same like upstairs

**Comment by paddatrapper** [comment]
Hmm... Yeah so the API needs to be generated and placed in the config file. A full config file example is here:

```
[general]
api_key = <random string of [A-Z|0-9] that is 20 characters long>
web_server_user = www-data
base_url = example.com
base_port = 80
base_dir = /
cache_ahead_hours = 1
airtime_dir =
station_id =
auth = local


[database]
host = localhost
dbname = airtime
dbuser = airtime
dbpass = airtime


[rabbitmq]
host = 127.0.0.1
port = 5672
user = airtime
password = airtime
vhost = /airtime


[current_backend]
storage_backend=file


[monit]
user =
password =



[pypo]
api_client = 'airtime'
cache_dir = '/var/tmp/airtime/pypo/cache/'
file_dir = '/var/tmp/airtime/pypo/files/' 
tmp_dir = '/var/tmp/airtime/pypo/tmp/'
cache_base_dir = '/var/tmp/airtime/pypo'
log_base_dir = '/var/log/airtime'
pypo_log_dir = '/var/log/airtime/pypo'
liquidsoap_log_dir = '/var/log/airtime/pypo-liquidsoap'
ls_host = '127.0.0.1'
ls_port = '1234'
poll_interval = 3600
push_interval = 1
cue_style = 'pre'
record_bitrate = 256
record_samplerate = 44100
record_channels = 2
record_sample_size = 16
record_file_type = 'ogg'
base_recorded_files = '/var/tmp/airtime/show-recorder/'


[soundcloud]
connection_retries = 3
time_between_retries = 60
soundcloud_client_id = 0
soundcloud_client_secret = 0
soundcloud_redirect_uri = http://libretime.example.org/soundcloud_callback.php
[facebook]
facebook_app_id = 0
facebook_app_url = http://example.org
facebook_app_api_key = 0

[ldap]
hostname = ldap.example.org
binddn = 'uid=libretime,cn=sysaccounts,cn=etc,dc=int,dc=example,dc=org'
password = hackme
account_domain = INT.EXAMPLE.ORG
basedn = 'cn=users,cn=accounts,dc=int,dc=example,dc=org'
filter_field = uid
groupmap_guest = 'cn=guest,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_host = 'cn=host,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_program_manager = 'cn=program_manager,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_admin = 'cn=admins,cn=groups,cn=accounts,dc=int,dc=example,dc=org'
groupmap_superadmin = 'cn=superadmin,cn=groups,cn=accounts,dc=int,dc=example,dc=org'

**Comment by nscpro** [comment]
But something is not good, i don't know why installation configuration stack. If i install Libretime on my raspberry, all works, but my VPS Ubuntu 18.04 LTS not work.

**Comment by paddatrapper** [comment]
I'm sure that is because of the port change

**Comment by nscpro** [comment]
should i somehow be able to install it in this way to change the ports after installation

**Comment by paddatrapper** [comment]
I think so

**Comment by nscpro** [comment]
okay i try that someway :)

**Comment by nscpro** [comment]
I test that over port 80 and same problem RabbitMQ internal error 500.

**Comment by paddatrapper** [comment]
What are the contents of the RabbitMQ logs (`/var/log/rabbitmq/`)?

**Comment by nscpro** [comment]
i see 4 different log files. 

rabbit@mail.log
rabbit@mail-sasl.log
startup_err
startup_log

**Comment by paddatrapper** [comment]
Post any that have potentially useful data

**Comment by nscpro** [comment]
![Screen Shot 2020-08-11 at 18 18 06](https://user-images.githubusercontent.com/16063685/89915624-42146e00-dbff-11ea-9d10-8a5f2e990f90.png)
![Screen Shot 2020-08-11 at 18 19 19](https://user-images.githubusercontent.com/16063685/89915647-4b9dd600-dbff-11ea-9d23-a823e471e0b4.png)

**Comment by paddatrapper** [comment]
Are you running into #889 perhaps?

**Comment by nscpro** [comment]
No. RabbitMQ password is default - airtime. I'm not changing that.

**Comment by paddatrapper** [comment]
Then I have no idea what is happening here...

**Comment by nscpro** [comment]
I think problem is libertime RabbitMQSetup file or somewhere. My RabbitMQ Management webpage woking and i can login using airtime user.

I don't understand what contains this "setup/setup-functions.php?obj=RabbitMQSetup" but somewhere have a problem

**Comment by nscpro** [comment]
I think i found problem, but this must be fix..

If already php7.3 is installed. Libretime install file not support Ubuntu Bionic 7.3 conf folder. This use only php7.2 not php7.3 default.

**Comment by paddatrapper** [comment]
Did you specifically install php 7.3?

**Comment by nscpro** [comment]
No, since i am using directadmin, when i install directadmin it automatically installed php7.3 version

**Comment by nscpro** [comment]
in short, Libretime installation does not allow php7.3, but version 7.2, so it installs all PHP modules for php7.2, and if php7.3 is used, these modules are not installed or enabled, which is why it does not allow the configuration page. go ahead with the installation..it needs to be fixed ... then everything will work too i think.

**Comment by paddatrapper** [comment]
Ok. This makes sense then. I think I am going to open a new issue tracking this though, as there is a lot of debugging in this one that will make following the discussion difficult.
